### PR TITLE
Extend deserial filter

### PR DESCRIPTION
**Pull request by paganpasta**

### CODE REVIEW

File: docs/api/utilities/serialisation.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Serialisation
 
+::: equinox.default_serialise_filter_spec
+
+---
+
+::: equinox.default_deserialise_filter_spec
+
+---
+
```
Comment (patrick-kidger) [inline docs/api/utilities/serialisation.md:10]: Can these go at the bottom of this page, as the other material on this page is more important.

File: equinox/serialisation.py [hunk 2]
```diff
@@ -144,6 +204,7 @@ def tree_deserialise_leaves(
         value from `like`).
     - `is_leaf`: Called on every node of `like`; if `True` then this node will be
         treated as a leaf.
+            print('_deserialize', spec, x)
```
Comment (patrick-kidger) [inline equinox/serialisation.py:207]: ?

File: equinox/tree.py [hunk 3]
```diff
@@ -118,6 +118,7 @@ def loss(model, ...):
     node_or_nodes = where(pytree)
     leaves1, structure1 = jax.tree_flatten(node_or_nodes_nowrapper, is_leaf=is_leaf)
     leaves2, structure2 = jax.tree_flatten(node_or_nodes)
+    print(leaves1, leaves2, "leaves 1 2")
```
Comment (patrick-kidger) [inline equinox/tree.py:121]: I assume this wasn't meant to be submitted!

File: tests/test_serialisation.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 import numpy as np
 
 import equinox as eqx
+import equinox.serialisation
```
Comment (patrick-kidger) [inline tests/test_serialisation.py:5]: It should be possible to access stuff just via `eqx`. (Or more broadly also `eqx.nn` and `eqx.experimental`.)

**Review by patrick-kidger — COMMENTED** [review]
Nits aside, LGTM!

**Comment by patrick-kidger** [comment]
LGTM.

The current build system updates the docs on every merge to main, so I'm going to leave this unmerged until we're ready for a new release. (Unless you want a new release for this - we can do that too.)

**Comment by paganpasta** [comment]
Yeah, waiting sounds fine to me.

**Comment by patrick-kidger** [comment]
Setting up a `dev` branch so that we can merge both this and #158.

**Pull request merged**